src/main/java/ca/team2706/frc/robot/logging/Log.java

```diff
@@ -26,10 +29,33 @@ public static void init() {
         Log.i("Allocated: " + ((double) Runtime.getRuntime().totalMemory()) / (1024 * 1024) + "MB");
         Log.i("Available: " + ((double) Runtime.getRuntime().maxMemory()) / (1024 * 1024) + "MB");
 
+        logBuildInfo();
+
         Log.i("Teleop game specific message: " + DriverStation.getInstance().getGameSpecificMessage());
         Robot.setOnStateChange(Log::printRobotState);
     }
 
+    private static void logBuildInfo() {
+
+        Properties properties = new Properties();
+        try {
+            properties.load(Log.class.getResourceAsStream(BUILD_INFO_NAME));
+        } catch (Exception e) {
+            Log.w("Could not load build info");
+        }
+
+        Log.i("Project name: " + properties.getProperty("name", "unknown"));
+        Log.i("Build timestamp: " + properties.getProperty("timestamp", "unknown"));
+        Log.i("Commit hash: " + properties.getProperty("commit.hash", "unknown"));
+        Log.i("Commit name: " + properties.getProperty("commit.name", "unknown"));
+        Log.i("Commit branch: " + properties.getProperty("commit.branch", "unknown"));
+        Log.i("Modified since commit: " + properties.getProperty("commit.modified", "unknown"));
+
+        if (!properties.getProperty("commit.modified", "unknown").equalsIgnoreCase("false")) {
```

Comment:
More of a question than a problem, why not just do `properties.getProperty("commit.modified", "unknown").equalsIgnoreCase("true")`?

Reply:
It just depends on what situations the warning should print in. This way, if it's unsure whether it's true it will print the warning. The other way will only print the warning if it's sure that there have been modifications.

Reply:
That makes sense, thanks!